tests/testthat/test-derive_var_trtemfl.R

```diff
@@ -34,6 +34,44 @@ expected <- tibble::tribble(
 
 adae <- select(expected, -starts_with("TRTEM"))
 
+expected2 <- tribble(
+  ~USUBJID, ~ASTDTM, ~AENDTM, ~AEITOXGR, ~AETOXGR, ~AEGRPID, ~TRTEMFL, ~TRTEM2FL,
+  # before treatment
+  "1", "2021-12-13T20:15", "2021-12-15T12:45", "1", "1", "1", NA, NA,
+  "1", "2021-12-14T20:15", "2021-12-14T22:00", "1", "3", "1", NA, NA,
+  # starting before treatment and ending during treatment
+  "1", "2021-12-30T20:15", "2022-01-14T01:23", "3", "3", "2", NA, NA,
+  "1", "2022-01-05T20:00", "2022-06-01T11:00", "3", "1", "2", NA, NA,
+  "1", "2022-01-10T20:15", "2022-01-11T01:23", "3", "2", "2", "Y", "Y",
+  "1", "2022-01-13T20:15", "2022-03-01T01:23", "3", "1", "2", "Y", "Y",
+  # starting during treatment
+  "1", "2022-01-01T12:00", "2022-01-02T23:25", "4", "4", "3", "Y", "Y",
```

Comment:
> Could we add the left part of the following diagram during derivation?

@kaz462 - This record is following the left part of the diagram. Correct me if you think otherwise

Reply:
This record is correctly flagged as a new AE. If there are subsequent records (within TEAE window) in the same group, I think the entire episode of the AE should be flagged without comparing the severity, what do you think?

Reply:
> I only see it getting flagged if the first of the AEs under group_var is TEAE and subsequent records will also be flagged irrespective of worsening condition.

Yeah, this is what I was mentioning here.